lib/kafka/consumer.rb

```diff
@@ -305,7 +313,7 @@ def each_batch(min_bytes: 1, max_bytes: 10485760, max_wait_time: 1, automaticall
       )
 
       consumer_loop do
-        batches = fetch_batches
+        batches = @interceptors.on_consume(batches: fetch_batches)
```

Comment:
I still think this is not the proper place to call the interceptor, it should be called below, after the control  messages are filtered out.
The interceptor should receive a list of messages and not a list of batches that contains a list of messages.

See the KIP docs:
>  * This is called when the records are about to be returned to the client.

Reply:
the consumer interceptor now receives a batch